File: Helpers/src/main/java/com/dg/controls/ViewPagerEx.java
package com.dg.controls;

import android.content.Context;
import android.content.res.TypedArray;
import android.support.v4.view.ViewPager;
import android.util.AttributeSet;
import android.view.MotionEvent;

import com.dg.R;

/**
 * Created by Daniel Cohen Gindi (danielgindi@gmail.com)
 */

public class ViewPagerEx extends ViewPager
{
    boolean mSwipeEnabled = true;

    public ViewPagerEx(Context context)
    {
        super(context);
    }

    public ViewPagerEx(Context context, AttributeSet attrs)
    {
        super(context, attrs);

        TypedArray a = context.obtainStyledAttributes(attrs, R.styleable.ViewPager);
        try
        {
            mSwipeEnabled = a.getBoolean(R.styleable.ViewPager_swipeEnabled, true);
        }
        finally
        {
            a.recycle();
        }
    }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent event)
    {
        if (mSwipeEnabled)
        {
            return super.onInterceptTouchEvent(event);
        }

        // Never allow swiping to switch between pages
        return false;
    }

    @Override
    public boolean onTouchEvent(MotionEvent event)
    {
        if (mSwipeEnabled)
        {
            return super.onTouchEvent(event);
        }

        // Never allow swiping to switch between pages
        return false;
    }
}

File: Helpers/src/main/java/com/dg/controls/ViewPagerEx.kt
package com.dg.controls

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import androidx.viewpager.widget.ViewPager
import com.dg.R

@Suppress("unused")
class ViewPagerEx : ViewPager
{
    @Suppress("MemberVisibilityCanBePrivate")
    internal var mSwipeEnabled = true

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)
    {

        val styledAttributes = context.obtainStyledAttributes(attrs, R.styleable.ViewPagerEx)
        try
        {
            mSwipeEnabled = styledAttributes.getBoolean(R.styleable.ViewPagerEx_swipeEnabled, true)
        }
        finally
        {
            styledAttributes.recycle()
        }
    }

    override fun onInterceptTouchEvent(event: MotionEvent): Boolean
    {
        return if (mSwipeEnabled)
        {
            super.onInterceptTouchEvent(event)
        }
        else false

        // Never allow swiping to switch between pages
    }

    override fun onTouchEvent(event: MotionEvent): Boolean
    {
        return if (mSwipeEnabled)
        {
            super.onTouchEvent(event)
        }
        else false

        // Never allow swiping to switch between pages
    }
}
